### PR TITLE
Module-2 task

### PR DESCRIPTION
**Pull request by antonskrebetz**
# Module 2 Taks

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 	const [authors, setAuthors] = useState(mockedAuthorsList);
 	const [search, setSearch] = useState('');
-	const [isCourses, setIsCourses] = useState(true);
 	const [showSearchCourses, setShowSearchCourses] = useState(false);
 
-	const onClickAddNewCourse = () => {
-		setIsCourses(false);
-	};
-
 	const handleSearchInput = (value: string) => {
```
Comment (barys-liutkouski-epam) [inline src/App.tsx:24]: if a component handles routing - it shouldn't do anything else, no passing props, no state hooks, etc, reason: separation of concerns, which would be very visible when unit testing this component - mocking router when testing a search input would feel very uneasy
`<Route path='registration' element={<Registration />} />` is what every route should look like

File: src/components/AuthComponent/AuthComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Navigate, Outlet } from 'react-router-dom';
+
+const useAuth = () => {
+	const user = localStorage.getItem('access_token');
+	return user ? true : false;
```
Comment (barys-liutkouski-epam) [inline src/components/AuthComponent/AuthComponent.tsx]: it's not very clear from the name what this hook returns, my first guess would be the authenticated user object

for better clarity you could return an object
`return { isAuthenticated: !!user }`

File: src/components/CourseInfo/CourseInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+							{formatCreationDate(course?.creationDate)}
+						</div>
+					</div>
+					<div className={styles.propsBlock}>
+						<div className={styles.propsTitle}>Authors:</div>
+					</div>
+					{course?.authors.map((author) => {
+						const currentAuthor = authors.find((el) => el.id === author)?.name;
+						return (
+							<p key={uuidv4()} className={styles.propsAuthor}>
```
Comment (barys-liutkouski-epam) [inline src/components/CourseInfo/CourseInfo.tsx]: it's better to pass no key at all then to generate a new one every time

does `author` have an id? maybe `author` itself is unique?

File: src/components/AuthComponent/AuthComponent.tsx (hunk 2 above)
Comment (antonskrebetz) [reply src/components/AuthComponent/AuthComponent.tsx]: Fixed

File: src/components/CourseInfo/CourseInfo.tsx (hunk 3 above)
Comment (antonskrebetz) [reply src/components/CourseInfo/CourseInfo.tsx]: Fixed

**Pull request merged**